internal/internal_worker.go

```diff
@@ -786,24 +678,30 @@ argsLoop:
 }
 
 // encode single value(like return parameter).
-func (th *hostEnvImpl) encodeArg(arg interface{}) ([]byte, error) {
-	return th.encode([]interface{}{arg})
+func encodeArg(dc encoded.DataConverter, arg interface{}) ([]byte, error) {
+	if dc == nil {
+		return newDefaultDataConverter().ToData(arg)
+	}
+	return dc.ToData(arg)
 }
 
 // decode single value(like return parameter).
-func (th *hostEnvImpl) decodeArg(data []byte, to interface{}) error {
-	return th.decode(data, []interface{}{to})
+func decodeArg(dc encoded.DataConverter, data []byte, to interface{}) error {
+	if dc == nil {
```

Comment:
could you have unit test to cover input with multiple parameters case

Reply:
added.